govc/object/collect.go

```diff
@@ -272,62 +313,75 @@ func (cmd *collect) Run(ctx context.Context, f *flag.FlagSet) error {
 		return err
 	}
 
-	ref := methods.ServiceInstance
-	arg := f.Arg(0)
+	p := property.DefaultCollector(client)
+	filter := new(property.WaitFilter)
 
-	if len(cmd.kind) != 0 {
-		ref = client.ServiceContent.RootFolder
-	}
+	if cmd.raw == "" {
+		ref := methods.ServiceInstance
+		arg := f.Arg(0)
 
-	switch arg {
-	case "", "-":
-	default:
-		if !ref.FromString(arg) {
-			l, ferr := finder.ManagedObjectList(ctx, arg)
-			if ferr != nil {
-				return err
-			}
+		if len(cmd.kind) != 0 {
+			ref = client.ServiceContent.RootFolder
+		}
 
-			switch len(l) {
-			case 0:
-				return fmt.Errorf("%s not found", arg)
-			case 1:
-				ref = l[0].Object.Reference()
-			default:
-				return flag.ErrHelp
+		switch arg {
+		case "", "-":
+		default:
+			if !ref.FromString(arg) {
+				l, ferr := finder.ManagedObjectList(ctx, arg)
+				if ferr != nil {
+					return err
+				}
+
+				switch len(l) {
+				case 0:
+					return fmt.Errorf("%s not found", arg)
+				case 1:
+					ref = l[0].Object.Reference()
+				default:
+					return flag.ErrHelp
+				}
 			}
 		}
-	}
 
-	p := property.DefaultCollector(client)
-	filter := new(property.WaitFilter)
+		var props []string
+		if f.NArg() > 1 {
+			props = f.Args()[1:]
+			cmd.single = len(props) == 1
+		}
 
-	var props []string
-	if f.NArg() > 1 {
-		props = f.Args()[1:]
-		cmd.single = len(props) == 1
-	}
+		props, err = cmd.toFilter(f, props)
+		if err != nil {
+			return err
+		}
 
-	props, err = cmd.toFilter(f, props)
-	if err != nil {
-		return err
-	}
+		if len(cmd.kind) == 0 {
+			filter.Add(ref, ref.Type, props)
+		} else {
+			m := view.NewManager(client)
 
-	if len(cmd.kind) == 0 {
-		filter.Add(ref, ref.Type, props)
-	} else {
-		m := view.NewManager(client)
+			v, cerr := m.CreateContainerView(ctx, ref, cmd.kind, true)
+			if cerr != nil {
+				return cerr
+			}
 
-		v, err := m.CreateContainerView(ctx, ref, cmd.kind, true)
-		if err != nil {
+			defer v.Destroy(ctx)
+
+			for _, kind := range cmd.kind {
+				filter.Add(v.Reference(), kind, props, v.TraversalSpec())
+			}
+		}
+	} else {
```

Comment:
Not that most the diff above was just moving the existing Run() body into an `if`, so we can have this `else`..